### DISCUSSION TITLE
Using pgvector instead of local code calculation

### DISCUSSION THREAD

**Discussion by xiaolong7713** [exadel-inc/CompreFace#1062] [Ideas] [answered]
Hi Devs:

First of all, thank you for this fantastic software.

I have couple of questions：

1. I found this code in blow, A cache will be createed on first Request, but it will be slow during the first request and consume a lot of memory when there are many subject in one apiKey.
```java
    public EmbeddingCollection getOrLoad(final String apiKey) {

        var result = cache.getIfPresent(apiKey);

        if (result == null) {
            result = embeddingService.doWithEnhancedEmbeddingProjectionStream(apiKey, EmbeddingCollection::from);

            cache.put(apiKey, result);

            notifyCacheEvent("UPDATE", apiKey);
        }

        return result;
    }
```

The pgvector is Open-source vector similarity search for Postgres, Using pgvector instead of local code calculation, also Using indexes to improve Face recognition speed.

https://github.com/pgvector/pgvector
https://supabase.com/blog/openai-embeddings-postgres-vector
https://github.com/exadel-inc/CompreFace/issues/1051

2. admin ui show subject list of Face Collection, All subjects were returned in the results. When I added 5000 subjects, the UI and server were crashed.   I think one of the good ways is to paginate the results。

**Comment by pospielov** [comment, marked as answer]
1. We create a version of CompreFace with the support of the MilvusDB vector database. I believe it's even more powerful than the Postgres plugin. I have already tested 20mln examples on one machine. This is a private version, at least for now. If you are working on enterprise and want to receive it, you can contact our sales or CompreFace support.
2. Yes, it makes sense. I'll create a story for this.

**Reply by xiaolong7713** [reply]
Thank you for the feedback.
